### PR TITLE
Support comma-delimited string as labels in issue template

### LINKED ISSUE #17877: Issue template with empty labels value
### Gitea Version

1.15.6

### Git Version

2.34.1

### Operating System

FreeBSD

### How are you running Gitea?

I run Gitea in a FreeBSD jail, installed from the package.

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

-

### Description

Hey,

when using an issue template that has an empty labels string (which github accepts), gitea bails out at reading the template with an
```
2021/12/02 00:31:21 ...ules/context/repo.go:927:IssueTemplatesFromDefaultBranch() [D] ExtractMetadata: yaml: unmarshal errors:
          line 4: cannot unmarshal !!str `` into []string
```
error.

Upon investigating the source, this seems to be coming from the labels key having an empty value in the issue template:
```yaml
---
name: Bug report
about: Create a report to help us improve
title: "[BUG]"
labels: ''
assignees: ''

---
<!-- Please don't delete this template or we'll close your issue -->

- [ ] I have read and understood the [FAQ](https://gitea.ksol.io/karolyi/py3-validate-email/src/branch/master/FAQ.md)

**Describe the bug**

A clear and concise description of what the bug is.

**To Reproduce**

Steps to reproduce the behavior:

**My debug output**

Output from the debug run described in the FAQ:

**Expected behavior**

A clear and concise description of what you expected to happen.

**Please complete the following information:**
 - OS: [e.g. Linux, FreeBSD, Windows]
 - Flavor and Version [e.g. Debian 22, FreeBSD 12.2]
 - Your network environment (ISP provided home connecton, or testing from an actual whitelisted server)
 - Your exact `py3-validate-email` module version

**Additional context**

Add any other context about the problem here.
```

I will try and remove that key now and see if it works, but this behavior was accepted with github, whereas it isn't with gitea. A quick fix would be nice to have for it, so other people won't bump into the same problem wondering why issue templates doesn't work after migrating from github.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by wolfogre**
The [labels in issue YAML templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#top-level-syntax) can be a string array or a comma-delimited string, so a single string should be valid labels.

The old codes committed in #20987 ignore this, that's why the warning is displayed:

image

And it also fixed #17877.

### CODE REVIEW

File: modules/structs/issue.go [hunk 1]
```diff
@@ -143,14 +145,15 @@ type IssueFormField struct {
...
-	Labels   []string          `json:"labels" yaml:"labels"`
-	Ref      string            `json:"ref" yaml:"ref"`
-	Content  string            `json:"content" yaml:"-"`
-	Fields   []*IssueFormField `json:"body" yaml:"body"`
-	FileName string            `json:"file_name" yaml:"-"`
+	Name       string            `json:"name" yaml:"name"`
+	Title      string            `json:"title" yaml:"title"`
+	About      string            `json:"about" yaml:"about"` // Using "description" in a template file is compatible
+	Labels     []string          `json:"labels" yaml:"-"`
+	LabelsNode yaml.Node         `json:"-" yaml:"labels"`
```
Comment (wxiaoguang) [inline modules/structs/issue.go]: It doesn't look good.

Could it be replaced by:

```
type IssueTemplateLabes []string

...
    Labels   IssueTemplateLabes         `json:"labels" yaml:"labels"`
...

func (labels *IssueTemplateLabes) UnmarshalYAML(value *yaml.Node) {
  ...
}
```

?

File: modules/structs/issue.go [hunk 2]
```diff
@@ -143,14 +147,46 @@ type IssueFormField struct {
...
+	Fields   []*IssueFormField   `json:"body" yaml:"body"`
+	FileName string              `json:"file_name" yaml:"-"`
+}
+
+type IssueTemplateLabels []string
+
+func (labels *IssueTemplateLabels) UnmarshalYAML(value *yaml.Node) error {
+	if value.IsZero() {
+		return nil
+	}
```
Comment (wxiaoguang) [inline modules/structs/issue.go]: here, it should reset the field if the value in yaml is zero.

File: modules/structs/issue.go [hunk 3]
```diff
@@ -143,14 +147,46 @@ type IssueFormField struct {
...
+		return nil
+	case yaml.SequenceNode:
+		var strs []string
+		if err := value.Decode(&strs); err != nil {
+			return err
+		}
+		*labels = strs
+		return nil
+	}
+	return fmt.Errorf("line %d: cannot unmarshal %s%s into IssueTemplateLabels",
```
Comment (wxiaoguang) [inline modules/structs/issue.go]: `%s%s`? by purpose?

**Review by wxiaoguang — APPROVED** [review]
Generally LGTM

File: modules/structs/issue.go (hunk 3 above)
Comment (wolfogre) [reply modules/structs/issue.go]: Try to be consistent with the yaml error format:

https://github.com/go-yaml/yaml/blob/f6f7691b1fdeb513f56608cd2c32c51f8194bf51/decode.go#L362

Comment (wxiaoguang) [reply modules/structs/issue.go]: BUT:

image

Comment (wxiaoguang) [reply modules/structs/issue.go]: I would suggest to use `("%s %q", value.ShortTag(), value.Value)`

Comment (wolfogre) [reply modules/structs/issue.go]: I see, maybe `line %d: cannot unmarshal %s into IssueTemplateLabels` is good enough:

```text
line 3: cannot unmarshal !!map into IssueTemplateLabels
```

You can find it in the tests I just added.

Comment (wolfogre) [reply modules/structs/issue.go]: `("%s %q", value.ShortTag(), value.Value)` does work well, I don't know why, but the value is empty:

```yaml
labels:
  a: aa
  b: bb
```

```
line 3: cannot unmarshal !!map "" into IssueTemplateLabels
```

image

**Review by KN4CK3R — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**